src/fennel/utils.fnl

```diff
@@ -128,6 +128,13 @@ Optionally takes a target table to insert the mapped values into."
   (accumulate [max 0 k (pairs tbl)]
     (if (= :number (type k)) (math.max max k) max)))
 
+(fn every [predicate seq]
+  (each [_ item (ipairs seq)]
+        (when (not (predicate item))
+          ;; use escape hatch for early return optimization
+          (lua "return false")))
```

Comment:
Can you rewrite this as an `accumulate` with an `:until` clause to skip the early return? We really want to avoid the use of the `lua` form.

Also let's name it `every?` instead since it's a predicate.